pkg/audit/service.go

```diff
@@ -20,29 +24,85 @@ type Service struct {
 	statdb   *statdb
 }
 
+type servicer struct {
+	service *Service
+	ctx        context.Context
+	cancel     context.CancelFunc
+	errs       []error
+	interval time.Duration
+}
+
 // NewService instantiates a Service with access to a Cursor and Verifier
 func NewService(pointers pdbclient.Client, transport transport.Client, overlay overlay.Client, id provider.FullIdentity) *Service {
 	cursor := NewCursor(pointers)
 	verifier := NewVerifier(transport, overlay, id)
 	return &Service{Cursor: cursor, Verifier: verifier}
 }
+// Config contains configurable values for audit service
+type Config struct {
+	Pointers pdbclient.Client ``
+	Transport transport.Client ``
+	Overlay overlay.Client ``
+	ID provider.FullIdentity ``
+	Interval time.Duration `help:"how frequently segements should audited" default:"30s"`
+}
 
-// Run calls Cursor and Verifier to continuously request random pointers, then verify data correctness at
-// a random stripe within a segment
-func (service *Service) Run(ctx context.Context) (err error) {
-	// TODO(James): make this function run indefinitely instead of once
-	stripe, err := service.Cursor.NextStripe(ctx)
-	if err != nil {
-		return err
-	}
-	failedNodes, err := service.Verifier.verify(ctx, stripe.Index, stripe.Segment)
+func (c Config) initialize(ctx context.Context) (servicer, error) {
+	var s servicer
+	s.service = NewService(c.Pointers, c.Transport, c.Overlay, c.ID)
+	s.ctx, s.cancel = context.WithCancel(ctx)
+
+	return s, nil
+}
+
+// Run runs the repairer with the configured values
+func (c Config) Run(ctx context.Context, server *provider.Provider) (err error) {
+	s, err := c.initialize(ctx)
 	if err != nil {
 		return err
 	}
-	for _, fail := range failedNodes {
-		service.statdb.RecordFailedAudit(fail)
-	}
-	return nil
+
+	return s.Run(ctx)
+}
+
+// Run calls Cursor and Verifier to continuously request random pointers, then verify data correctness at
+// a random stripe within a segment
+func (s *servicer) Run(ctx context.Context) (err error) {
+	// TODO(James): make this function run indefinitely instead of once
+
+	defer mon.Task()(&ctx)(&err)
+
+	zap.S().Info("Audit cron is starting up")
+	ticker := time.NewTicker(s.interval)
+	defer ticker.Stop()
+
+	ctx, cancel := context.WithCancel(ctx)
+	defer cancel()
+
+	go func() {
+		for {
+			select {
+			case <- ticker.C:
+				stripe, err := s.service.Cursor.NextStripe(ctx)
+				if err != nil {
+					s.errs = append(s.errs, err)
+					s.cancel()
+				}
+				failedNodes, err := s.service.Verifier.verify(ctx, stripe.Index, stripe.Segment)
+				if err != nil {
+					s.errs = append(s.errs, err)
+					s.cancel()
+				}
+				for _, fail := range failedNodes {
+					s.service.statdb.RecordFailedAudit(fail)
```

Comment:
for the failed audits- what does 'fail' look like? are we going to be recording times audit failed, number nodes failed, etc?

Reply:
Hey, you can see more info about this in my pr: https://github.com/storj/storj/pull/467
This is how the statdb Node proto is updated when `RecordAuditFailed` is called:
```
&proto.Node{
			NodeId:             []byte(fail.GetId()),
			AuditSuccess:       false,
			IsUp:               true,
			UpdateLatency:      false,
			UpdateAuditSuccess: true,
			UpdateUptime:       true,
		}
```
Also yes, `total_audit_count`, `audit_success_count`, and `audit_success_ratio` are recorded in statdb.

Reply:
yup, yup :) just answered my question while reviewing your pr this morning!